### PR TITLE
#21 Add: general utils

### LINKED ISSUE #21: Extract URL de-fanging into a utils file

### PR DESCRIPTION
**Pull request by MaXeraph**
## Changes
- Added Defang utils
- Added directory and file reader utils

### CODE REVIEW

File: src/utils/file.utils.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * if paths is empty, __dirname is used
+ *
+ * @param paths A sequence of paths or path segments.
+ * @throws {TypeError} if any of the arguments is not a string.
+ */
+export function readDirectory(
+  fileExtension: string = ".ts",
+  ...paths: string[]
+) {
```
Comment (umar-ahmed) [inline src/utils/file.utils.ts:17]: This interface is a bit weird. I wonder if instead we use a glob matching library instead:

```ts
readDirectory('./src/commands/*.ts')
```

File: src/commands/check.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import axios from "axios";
 import { CommandInteraction, SlashCommandBuilder } from "discord.js";
 import { env } from "../env";
+import { defangUrl } from "../utils/url.utils";
```
[the remark this answers is not in the document]
Comment (MaXeraph) [reply src/commands/check.ts]: https://github.com/chainpatrol/discord-bot/pull/27/commits/74267962ccd179901109f3cf81d573171d8b99be

File: src/utils/file.utils.ts (hunk 1 above)
Comment (MaXeraph) [reply src/utils/file.utils.ts]: https://github.com/chainpatrol/discord-bot/pull/27/commits/eea90648b6d2bf148c1dc79edf7358feba98f4df

**Review by umar-ahmed — COMMENTED** [review]
code looks good

**Review by umar-ahmed — APPROVED** [review]
QA LGTM

**Pull request merged**